File: BankatiProject_Backend/services/TransactionService/src/main/java/com/example/transactionservice/transaction/response/TransactionResponse.java
package com.example.transactionservice.transaction.response;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record TransactionResponse (
        String id,
        BigDecimal amount,
        String transactionMethod,
        String beneficiaryId,
        String beneficiaryName,
        String beneficiaryPhone,
        String beneficiaryRole,
        String transactionType,
        String status,
        String senderCurrency,
        String beneficiaryCurrency,
        LocalDateTime createdDate,
        String senderId,
        String senderName,
        String senderPhoneNumber,
        String senderRole,
        LocalDateTime lastModifiedDate,
        LocalDateTime validatedDate
){
}

File: BankatiProject_Front-end/src/app/client/models/transactionResponse.ts

export interface TransactionResponse {

   id: string;
   amount: number;
   transactionMethod: string;
   beneficiaryId: string;
   beneficiaryName: string;
   beneficiaryPhone: string;
   beneficiaryRole: string;
   transactionType: string;
   status: string;
   senderCurrency: string;
   beneficiaryCurrency: string;
   createdDate: string;
   senderId: string;
   senderName: string;
   senderPhoneNumber: string;
   senderRole: string;
   lastModifiedDate: string;
   validatedDate: string;
}
